= = = = = = = = = = = = = = = = = = = =
## Starting Time: 09-09 13:15:09
Namespace(dataset='huggingface', llm='Mistral-7B', llm_model_path='', seed=0, device='cuda:0', max_txt_length=512, max_ans_length=512, gnn_in_dim=1024, gnn_hidden_dim=1024, gnn_output_dim=4096, n_layers=2, gnn_type='SAGE', max_degree=300, num_epochs=2, batch_size=6, eval_batch_size=6, patience=2, lr=1e-05, wd=0.05, output_dir='output', grad_steps=4, name='centrality')

[Training Data] # Chain Samples 1536 (51.20)
[Data Split] # Train 3000  # Test 500
# Train 2400   # Val 600   # Test 500
Loading checkpoint shards: 100%|███████████████████| 3/3 [00:01<00:00,  2.11it/s]
Finish loading pre-trained Mistral-7B model!
Trainable params 11107328 || all params 7252839424 || trainable% 0.15314
/nas/home/ktshim/tool/pool/graphtoken/../utils/datautil.py:24: RuntimeWarning: divide by zero encountered in power
  d_inv = np.power(row_sum, -0.5).flatten()
/nas/home/ktshim/tool/pool/graphtoken/../utils/datautil.py:19: UserWarning: torch.sparse.SparseTensor(indices, values, shape, *, device=) is deprecated.  Please use torch.sparse_coo_tensor(indices, values, shape, dtype=, device=). (Triggered internally at /pytorch/torch/csrc/utils/tensor_new.cpp:653.)
  return torch.sparse.FloatTensor(index, data, torch.Size(coo.shape))
  0%|                                                    | 0/800 [00:00<?, ?it/s]E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0] failed while attempting to run meta for aten.add.Tensor
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0] Traceback (most recent call last):
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_subclasses/fake_tensor.py", line 2717, in _dispatch_impl
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     r = func(*args, **kwargs)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]         ^^^^^^^^^^^^^^^^^^^^^
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_ops.py", line 829, in __call__
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     return self._op(*args, **kwargs)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]            ^^^^^^^^^^^^^^^^^^^^^^^^^
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_prims_common/wrappers.py", line 309, in _fn
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     result = fn(*args, **kwargs)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]              ^^^^^^^^^^^^^^^^^^^
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_compile.py", line 53, in inner
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     return disable_fn(*args, **kwargs)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/eval_frame.py", line 929, in _fn
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     return fn(*args, **kwargs)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]            ^^^^^^^^^^^^^^^^^^^
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_prims_common/wrappers.py", line 149, in _fn
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     result = fn(**bound.arguments)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]              ^^^^^^^^^^^^^^^^^^^^^
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_refs/__init__.py", line 1122, in add
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     a, b = _maybe_broadcast(a, b)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]            ^^^^^^^^^^^^^^^^^^^^^^
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_refs/__init__.py", line 429, in _maybe_broadcast
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     common_shape = _broadcast_shapes(
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]                    ^^^^^^^^^^^^^^^^^^
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_refs/__init__.py", line 417, in _broadcast_shapes
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     torch._check(
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/__init__.py", line 1684, in _check
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     _check_with(RuntimeError, cond, message)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]   File "/home/ktshim/.local/lib/python3.12/site-packages/torch/__init__.py", line 1666, in _check_with
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0]     raise error_type(message_evaluated)
E0909 14:15:27.029000 3464958 torch/_subclasses/fake_tensor.py:2721] [7/0] RuntimeError: Attempting to broadcast a dimension of length 1024 at -1! Mismatching argument at index 1 had torch.Size([23, 1024]); but expected shape should be broadcastable to [23, 4096]
Traceback (most recent call last):
  File "/nas/home/ktshim/tool/pool/graphtoken/main.py", line 137, in <module>
    loss, logits = model(batch, task_graph)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/eval_frame.py", line 375, in __call__
    return super().__call__(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1773, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1784, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/eval_frame.py", line 736, in compile_wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1773, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1784, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nas/home/ktshim/tool/pool/graphtoken/glm_node_centrality.py", line 87, in forward
    requests = self.tokenizer(samples["request"], add_special_tokens=False)
  File "/nas/home/ktshim/tool/pool/graphtoken/glm_node_centrality.py", line 88, in torch_dynamo_resume_in_forward_at_87
    labels = self.tokenizer(samples["label"], add_special_tokens=False)
  File "/nas/home/ktshim/tool/pool/graphtoken/glm_node_centrality.py", line 91, in torch_dynamo_resume_in_forward_at_88
    eos_tokens = self.tokenizer(self.EOS, add_special_tokens=False)
  File "/nas/home/ktshim/tool/pool/graphtoken/glm_node_centrality.py", line 92, in torch_dynamo_resume_in_forward_at_91
    eos_user_tokens = self.tokenizer(self.EOS_USER, add_special_tokens=False)
  File "/nas/home/ktshim/tool/pool/graphtoken/glm_node_centrality.py", line 93, in torch_dynamo_resume_in_forward_at_92
    bos_embeds = self.word_embedding(self.tokenizer(self.BOS, add_special_tokens=False, return_tensors='pt').input_ids[0].to(self.device))
  File "/nas/home/ktshim/tool/pool/graphtoken/glm_node_centrality.py", line 98, in torch_dynamo_resume_in_forward_at_93
    node_embeds = self.encode_task_graph(task_graph, batch_size)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 1495, in __call__
    return self._torchdynamo_orig_callable(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 1272, in __call__
    result = self._inner_convert(
             ^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 629, in __call__
    return _compile(
           ^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 1111, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_utils_internal.py", line 97, in wrapper_function
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 793, in compile_inner
    return _compile_inner(code, one_graph, hooks, transform)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 832, in _compile_inner
    out_code = transform_code_object(code, transform)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/bytecode_transformation.py", line 1424, in transform_code_object
    transformations(instructions, code_options)
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 267, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 753, in transform
    tracer.run()
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 3497, in run
    super().run()
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 1363, in run
    while self.step():
          ^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 1267, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 2851, in BINARY_OP
    return _binary_op_lookup[inst.arg](self, inst)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 425, in impl
    self.push(fn_var.call_function(self, self.popn(nargs), {}))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/variables/builtin.py", line 1189, in call_function
    return handler(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/variables/builtin.py", line 1149, in _handle_insert_op_in_graph
    return wrap_fx_proxy(tx, proxy)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/variables/builder.py", line 2559, in wrap_fx_proxy
    return wrap_fx_proxy_cls(target_cls=TensorVariable, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/variables/builder.py", line 2625, in wrap_fx_proxy_cls
    return _wrap_fx_proxy(
           ^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/variables/builder.py", line 2723, in _wrap_fx_proxy
    example_value = get_fake_value(proxy.node, tx, allow_non_graph_fake=True)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/utils.py", line 3355, in get_fake_value
    raise TorchRuntimeError(str(e)).with_traceback(e.__traceback__) from None
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/utils.py", line 3253, in get_fake_value
    ret_val = wrap_fake_exception(
              ^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/utils.py", line 2753, in wrap_fake_exception
    return fn()
           ^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/utils.py", line 3254, in <lambda>
    lambda: run_node(tx.output, node, args, kwargs, nnmodule)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/utils.py", line 3462, in run_node
    raise RuntimeError(make_error_message(e)).with_traceback(
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/utils.py", line 3421, in run_node
    return node.target(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/utils/_stats.py", line 28, in wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_subclasses/fake_tensor.py", line 1352, in __torch_dispatch__
    return self.dispatch(func, types, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_subclasses/fake_tensor.py", line 2058, in dispatch
    return self._cached_dispatch_impl(func, types, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_subclasses/fake_tensor.py", line 1487, in _cached_dispatch_impl
    output = self._dispatch_impl(func, types, args, kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_subclasses/fake_tensor.py", line 2717, in _dispatch_impl
    r = func(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_ops.py", line 829, in __call__
    return self._op(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_prims_common/wrappers.py", line 309, in _fn
    result = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_compile.py", line 53, in inner
    return disable_fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_dynamo/eval_frame.py", line 929, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_prims_common/wrappers.py", line 149, in _fn
    result = fn(**bound.arguments)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_refs/__init__.py", line 1122, in add
    a, b = _maybe_broadcast(a, b)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_refs/__init__.py", line 429, in _maybe_broadcast
    common_shape = _broadcast_shapes(
                   ^^^^^^^^^^^^^^^^^^
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/_refs/__init__.py", line 417, in _broadcast_shapes
    torch._check(
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/__init__.py", line 1684, in _check
    _check_with(RuntimeError, cond, message)
  File "/home/ktshim/.local/lib/python3.12/site-packages/torch/__init__.py", line 1666, in _check_with
    raise error_type(message_evaluated)
torch._dynamo.exc.TorchRuntimeError: Dynamo failed to run FX node with fake tensors: call_function <built-in function add>(*(FakeTensor(..., device='cuda:0', size=(23, 4096), grad_fn=<AddBackward0>), FakeTensor(..., device='cuda:0', size=(23, 1024), grad_fn=<EmbeddingBackward0>)), **{}): got RuntimeError('Attempting to broadcast a dimension of length 1024 at -1! Mismatching argument at index 1 had torch.Size([23, 1024]); but expected shape should be broadcastable to [23, 4096]')

from user code:
   File "/nas/home/ktshim/tool/pool/graphtoken/glm_node_centrality.py", line 68, in encode_task_graph
    node_embeds = node_embeds + in_degree_embed + out_degree_embed

Set TORCHDYNAMO_VERBOSE=1 for the internal stack trace (please do this especially if you're reporting a bug to PyTorch). For even more developer context, set TORCH_LOGS="+dynamo"
